src/mlpack/tests/ne_test.cpp

```diff
@@ -0,0 +1,238 @@
+ */
+BOOST_AUTO_TEST_CASE(NEGenomeTest)
+{
+  mlpack::math::RandomSeed(1);
+
+  // Construct seed genome for xor task.
+  int id = 0;
+  int numInput = 3;
+  int numOutput = 1;
+  double fitness = -1;
+  std::vector<NeuronGene> neuronGenes;
+  std::vector<LinkGene> linkGenes;
+
+  NeuronGene inputGene1(0, INPUT, LINEAR, 0, std::vector<double>(), 0, 0);
+  NeuronGene inputGene2(1, INPUT, LINEAR, 0, std::vector<double>(), 0, 0);
+  NeuronGene biasGene(2, BIAS, LINEAR, 0, std::vector<double>(), 0, 0);
+  NeuronGene outputGene(3, OUTPUT, SIGMOID, 1, std::vector<double>(), 0, 0);
+  NeuronGene hiddenGene(4, HIDDEN, SIGMOID, 0.5, std::vector<double>(), 0, 0);
+
+  neuronGenes.push_back(inputGene1);
+  neuronGenes.push_back(inputGene2);
+  neuronGenes.push_back(biasGene);
+  neuronGenes.push_back(outputGene);
+  neuronGenes.push_back(hiddenGene);
+
+  LinkGene link1(0, 3, 0, 1, true);
+  LinkGene link2(1, 3, 0, 1, true);
+  LinkGene link3(2, 3, 0, 0.5, true);
+  LinkGene link4(0, 4, 0, -0.5, true);
+  LinkGene link5(1, 4, 0, 2, true);
+  LinkGene link6(2, 4, 0, 1, true);
+  LinkGene link7(4, 3, 0, 0.1, true);
+
+  linkGenes.push_back(link1);
+  linkGenes.push_back(link2);
+  linkGenes.push_back(link3);
+  linkGenes.push_back(link4);
+  linkGenes.push_back(link5);
+  linkGenes.push_back(link6);
+  linkGenes.push_back(link7);
+
+  Genome seedGenome = Genome(0, 
+                             neuronGenes,
+                             linkGenes,
+                             numInput,
+                             numOutput,
+                             fitness);
+
+  // Test seed genome.
+  std::vector<std::vector<double>> inputs;  // TODO: use arma::mat for input.
+  std::vector<double> input1 = {0, 0, 1};
+  std::vector<double> input2 = {0, 1, 1};
+  std::vector<double> input3 = {1, 0, 1};
+  std::vector<double> input4 = {1, 1, 1};
+  inputs.push_back(input1);
+  inputs.push_back(input2);
+  inputs.push_back(input3);
+  inputs.push_back(input4);
+
+  std::vector<double> outputs;
+  outputs.push_back(1);
+  outputs.push_back(0);
+  outputs.push_back(1);
+  outputs.push_back(0);
+
+  for (int i = 0; i < 4; ++i) {
+    seedGenome.Activate(inputs[i]);
+    std::vector<double> output;
+    seedGenome.Output(output);
+    std::cout << "Output is: " << output[0] << std::endl;
+  }
+}
+
+/**
+ * Test Species class.
+ */
+BOOST_AUTO_TEST_CASE(NESpeciesTest)
+{
+
+}
+
+/**
+ * Test CNE by XOR task.
+ */
+BOOST_AUTO_TEST_CASE(NECneXorTest)
+{
+  mlpack::math::RandomSeed(1);
+
+  // Set CNE algorithm parameters.
+  Parameters params;
+  params.speciesSize = 500;
+  params.mutateRate = 0.1;
+  params.mutateSize = 0.02;
+  params.elitePercentage = 0.2;
+  params.maxGeneration = 5000;
+
+  // Construct seed genome for xor task.
+  int id = 0;
+  int numInput = 3;
+  int numOutput = 1;
+  double fitness = -1;
+  std::vector<NeuronGene> neuronGenes;
+  std::vector<LinkGene> linkGenes;
+
+  NeuronGene inputGene1(0, INPUT, LINEAR, 0, std::vector<double>(), 0, 0);
```

Comment:
Do you think, we could set a default argument for the  ``coordinate``parameter. That way we don't have to specify ``std::vector<double>()`` every time we create a new NeuronGene.